cns/service/main.go

```diff
@@ -405,6 +410,25 @@ func main() {
 	configuration.SetCNSConfigDefaults(&cnsconfig)
 	logger.Printf("[Azure CNS] Read config :%+v", cnsconfig)
 
+	// We might be configured to reinitialize state from the CNI instead of the apiserver.
+	// If so, we should check that the the CNI is new enough to support the state commands,
+	// otherwise we fall back to the existing behavior.
+	if cnsconfig.InitializeFromCNI {
+		isGoodVer, err := cni.IsDumpStateVer()
```

Comment:
This is only checked if ChannelMode is cns.CRD.

Reply:
For other configurations (like for ACI), reconcilation happens between DNC <> CNS